### DISCUSSION TITLE
How to sync the siblings' states

### DISCUSSION THREAD

**Discussion by atacan** [pointfreeco/swift-composable-architecture#2791] [Q&A]
What's the best way to sync the state of a child based on the state of its sibling? 

I thought it's not a good idea to look at the actions of the sibling in the parent and update the state of the child accordingly, because then we keep replicating the business logic of the sibling in two places. 

I added an `onReceive` modifier to the child view where I used the Combine publisher of the sibling's state to change the child's state inside the parent reducer. Is there a better way?

Example:

Parent has two independent children.
When the state of Child A's button becomes "enabled" then I want to make the state of Child B's text "Disabled".

**Comment by mbrandonw** [comment]
Hi @atacan, there are a lot of discussions about "sharing state" that you can find in here for techniques to do this. And very soon we will be releasing official tools for doing this. We will even give a sneak peek of the tools in our [livestream](https://www.pointfree.co/live) tomorrow. 🤫
